data-serving/scripts/aggregate/aggregate/app.py

```diff
@@ -110,19 +110,22 @@ def get_jhu_counts():
 
     Return aggregated counts by country as Series.
     """
-
+    logging.debug("Getting aggregated counts from JHU")
     now = datetime.datetime.now().strftime("%m-%d-%Y")
     date = datetime.datetime.now()
     url = f"https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_daily_reports/{now}.csv"
     req = requests.head(url, timeout=10)
 
-    while req.status_code != 200:
+    retries = 5
+    current_attempt = 0
+    while req.status_code != 200 and current_attempt < retries:
         logging.info("Got status " + str(req.status_code) + " for '" + url + "'")
         date = date - datetime.timedelta(days=1)
         now = date.strftime("%m-%d-%Y")
         logging.info(f"Checking for JHU data on {now}")
         url = f"https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_daily_reports/{now}.csv"
         req = requests.head(url, timeout=10)
+        current_attempt += 1
```

Comment:
Two things here, one is definitely a problem and the other I don't understand.

1. This loop doesn't handle the failure condition at all. If the repo renames master to main, or deletes this file, or GitHub goes down, this will just receive an error five times and then carry on as normal. There should definitely be an early exit here if we don't actually get the data. Looking at the rest of the code it would be possible to return an empty Pandas series and then not have JHU data downstream, but that might not be desirable.
2. I don't see the value in testing availability with a HEAD request if we are going to unconditionally GET the data anyway, which is the thing I'm confused by. I would use HEAD if we expected a large response and wanted to test whether we could use cached data, or needed to pre-allocate resources before the download. As it is, this is testing that we _could_ download the data, then making a separate request to download it (which could still fail). I _think_ it's better to just try grabbing the data with a GET in this loop and using it if successful: unless there's another reason to prefer HEAD that I missed.

Reply:
Yes, and I added an early exit. New logic is: check for today's file. If it does not exist, check for yesterday's file. If that does not exist, then exit. Good point about the HEAD request being redundant. I agree, the file is only ~500 kb. I removed those and use GETs now.